### ISSUE TITLE
DOC: Latest docs stable build doesn't include bugfix release notes

### ISSUE THREAD

**Issue by asishm** [numpy/numpy#26613]
### Issue with current documentation:

When I go to release notes, they stop at 1.26.0 (and doesn't show 1.26.1 through 1.26.4) at https://numpy.org/doc/stable/release.html However switching to the dev dropdown does show the other 1.26.x releases as well as the 2.x one.

Apologies if I'm misinterpreting what the UI should be showing or maybe this is how the numpy docs always behaved and I never noticed. 

![image](https://github.com/numpy/numpy/assets/1978117/7563a9c4-083c-4bcf-a57f-bb17b05380f8)

**Comment by ngoldbaum** [comment]
You can see them in the devdocs: https://numpy.org/devdocs/release.html

I suspect the docs were never rebuilt for each minor release.

**Comment by asishm** [comment]
Yup, that's what clicking the dev option does on the version switcher. I just wanted to raise the issue because people may think that v1.26.1 - v.1.26.4 are somehow "unstable" because they don't show up under the stable docs (but instead show up under the dev docs)

**Comment by charris** [comment]
> I suspect the docs were never rebuilt for each minor release.

It is rare to rebuild the docs for minor releases, I only recall doing it once. Because the other minor releases point to the relevant release notes, I suspect a maintenance lapse on the numpy.org site.

**Comment by ngoldbaum** [comment]
Ah I see, yes there should be builds for each minor release.

**Comment by charris** [comment]
> there should be builds for each minor release.

Are we talking minor or micro. Docs get built for the x.x.0 releases, but not the bug fix releases. It is the bug fix releases that are missing and all they get is a note in the news section.

**Comment by charris** [comment]
@ngoldbaum Still not quite right. Let's start with semantic versioning definitions, the version tag looks like: `major.minor.micro`. I only build the docs when micro is 0, you can see this documented in `doc/RELEASE_WALKTHROUGH.rst`. Note that `patch` is often used in place of `micro`, but the meaning is the same.

**Comment by ngoldbaum** [comment]
I think that might be an overly strict view of semver. Effectively, for numpy, the "micro" you're referring to are bugfix releases and the "minor" releases are major releases. IMO of course.

**Comment by mattip** [comment]
I think it makes sense to rebuild documentation for all releases on PyPI. Would it be a large burden to do that? I think the version dropdown (in the upper right corner) can continue to only have minor versions, no micro versions needed. When installing from PyPI, the user will get 1.26.4, so that should be the documentation linked to via `stable`, and it should show the release notes and changes since 1.26.0

**Comment by charris** [comment]
> I think it makes sense to rebuild documentation for all releases on PyPI.

It is unusual to backport any changes to the documentation for micro releases.  The dropdown menu in question is for the release notes, not the documentation. That could be updated along with the announcement of the release, but it seems to be a new addition and that is probably why it cuts off where it does.

**Comment by mattip** [comment]
The important change to documentation with a micro release is the notes. The documents built for the 1.26.0 release cannot display the release notes for 1.26.4. What do we expect users to see when they look at the `stable` documentation's release notes?

**Comment by asishm** [comment]
Thanks everyone for the responses. I understand the current behavior and it makes sense now given the context that the docs don't get rebuilt for a patch release.

So that explains to me why 1.25.x (other than 1.25.0) release notes only appear under 1.26 and higher. Similarly, with 1.26.0 being the most recent minor release, the patch version release notes would show up under when dev is selected in the version switcher.

**Comment by charris** [comment]
> notes only appear under 1.26 and higher.

I think whoever added the drop down list made it complete up to the time it was added and it hasn't been updated since.

**Comment by mattip** [comment]
We update that site only on major releases, mainly because creating the documentation for every release would blow up the storage required. But the release notes are available on [the dev version](https://numpy.org/devdocs/release.html) of the documentation that gets rebuilt from HEAD on every merge-to-main.

Closing as not planned.
